pkg/healthcheck/healthcheck.go

```diff
@@ -1364,6 +1370,35 @@ func (hc *HealthChecker) allCategories() []Category {
 	}
 }
 
+// CheckProxyVersionsUpToDate checks if all the proxies are on the latest
+// installed version
+func (hc *HealthChecker) CheckProxyVersionsUpToDate(pods []corev1.Pod) error {
+	outdatedPods := []string{}
+	for _, pod := range pods {
+		proxyVersion := k8s.GetProxyVersion(pod)
+		if err := hc.latestVersions.Match(proxyVersion); err != nil {
+			outdatedPods = append(outdatedPods, fmt.Sprintf("\t* %s (%s)", pod.Name, proxyVersion))
+		}
+	}
+	if len(outdatedPods) > 0 {
+		podList := strings.Join(outdatedPods, "\n")
+		return fmt.Errorf("Some data plane pods are not running the current version:\n%s", podList)
+	}
+	return nil
+}
+
+// CheckIfProxyVersionsMatchWithCLI checks if the latest proxy version
+// matches that of the CLI
+func CheckIfProxyVersionsMatchWithCLI(pods []corev1.Pod) error {
```

Comment:
Is there a reason why this method is not attached to `hc *HealthChecker` while `CheckProxyVersionsUpToDate` is?

Reply:
My thinking here was that this method does need `hc` anywhere for its functionality and hence did not add 🤔

Reply:
Ah, I see. That makes sense. 👍